### PR TITLE
Backports 0.15 pr9

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
will be rebased after pr8 is merged

review https://github.com/dashpay/dash/pull/2771/commits/070b209c50b304f335d751781c2de43e5f8e6c5f and on

### CODE REVIEW

**Comment by UdjinM6** [comment]
needs rebase

**Comment by codablock** [comment]
I'd suggest to first rebase https://github.com/dashpay/dash/pull/2912 so that we can merge it first, and only then rebase this PR.

**Comment by PastaPastaPasta** [comment]
rebased :+1:

**Comment by UdjinM6** [comment]
build fails https://travis-ci.org/dashpay/dash/jobs/534614270#L2085

File: test/functional/assumevalid.py [hunk 1]
```diff
@@ -166,25 +186,19 @@ def run_test(self):
...
+        # Send blocks to node0. Block 102 will be rejected.
+        self.send_blocks_until_disconnected(node0)
+        self.assert_blockchain_height(self.nodes[0], 101)
 
-        # Send 3102 blocks to node1. All blocks will be accepted.
+        # Send all blocks to node1. All blocks will be accepted.
         for i in range(2202):
             node1.send_message(msg_block(self.blocks[i]))
-        node1.sync_with_ping() # make sure the most recent block is synced
+        node1.sync_with_ping()  # make sure the most recent block is synced
```
Comment (UdjinM6) [inline test/functional/assumevalid.py]: needs https://github.com/bitcoin/bitcoin/pull/10114 as mentioned in the original PR

File: src/init.cpp [hunk 2]
```diff
@@ -646,6 +646,7 @@ std::string HelpMessage(HelpMessageMode mode)
     strUsage += HelpMessageOpt("-rpcauth=<userpw>", _("Username and hashed password for JSON-RPC connections. The field <userpw> comes in the format: <USERNAME>:<SALT>$<HASH>. A canonical python script is included in share/rpcuser. The client then connects normally using the rpcuser=<USERNAME>/rpcpassword=<PASSWORD> pair of arguments. This option can be specified multiple times"));
     strUsage += HelpMessageOpt("-rpcport=<port>", strprintf(_("Listen for JSON-RPC connections on <port> (default: %u or testnet: %u)"), BaseParams(CBaseChainParams::MAIN).RPCPort(), BaseParams(CBaseChainParams::TESTNET).RPCPort()));
     strUsage += HelpMessageOpt("-rpcallowip=<ip>", _("Allow JSON-RPC connections from specified source. Valid for <ip> are a single IP (e.g. 1.2.3.4), a network/netmask (e.g. 1.2.3.4/255.255.255.0) or a network/CIDR (e.g. 1.2.3.4/24). This option can be specified multiple times"));
+    strUsage += HelpMessageOpt("-rpcserialversion", strprintf(_("Sets the serialization of raw transaction or block hex returned in non-verbose mode, non-segwit(0) or segwit(1) (default: %d)"), DEFAULT_RPC_SERIALIZE_VERSION));
```
Comment (UdjinM6) [inline src/init.cpp]: We don't have `-rpcserialversion`, this should be dropped

File: test/functional/sendheaders.py [hunk 3]
```diff
@@ -92,9 +92,9 @@
 
 direct_fetch_response_time = 0.05
 
-class BaseNode(SingleNodeConnCB):
+class BaseNode(NodeConnCB):
```
Comment (UdjinM6) [inline test/functional/sendheaders.py]: Should also fix `dip4-coinbasemerkleroots.py` and `llmq-is-cl-conflicts.py` the same way as it's done here

File: src/qt/test/wallettests.cpp [hunk 4]
```diff
@@ -22,7 +22,9 @@ namespace
 //! Press "Yes" button in modal send confirmation dialog.
 void ConfirmSend()
 {
-    QTimer::singleShot(0, Qt::PreciseTimer, []() {
+    QTimer* timer = new QTimer;
+    timer->setSingleShot(true);
+    QObject::connect(timer, &QTimer::timeout, []() {
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/qt/test/wallettests.cpp]: https://github.com/dashpay/dash/pull/2771/commits/dba492d2745122e21f58b90e0e570445731aa85b

File: test/functional/assumevalid.py (hunk 1 above)
Comment (PastaPastaPasta) [reply test/functional/assumevalid.py]: https://github.com/dashpay/dash/pull/2771/commits/b0b1bbe90b8bff12265fe9dc16d0017fa09a2b34

File: src/init.cpp (hunk 2 above)
Comment (PastaPastaPasta) [reply src/init.cpp]: https://github.com/dashpay/dash/pull/2771/commits/f94766480bf1c42fd40442829938d3152a079b9e

File: test/functional/sendheaders.py (hunk 3 above)
Comment (PastaPastaPasta) [reply test/functional/sendheaders.py]: https://github.com/dashpay/dash/pull/2771/commits/50a99de1f09f9bcfa242c551028ce7759a689639

File: test/functional/dip4-coinbasemerkleroots.py [hunk 5]
```diff
@@ -16,9 +16,9 @@
 
 '''
 
-class TestNode(SingleNodeConnCB):
+class TestNode(NodeConnCB):
     def __init__(self):
-        SingleNodeConnCB.__init__(self)
+        NodeConnCB.__init__(self)
```
Comment (UdjinM6) [inline test/functional/dip4-coinbasemerkleroots.py]:
```suggestion
        super().__init__()
```

File: test/functional/llmq-is-cl-conflicts.py [hunk 6]
```diff
@@ -15,9 +15,9 @@
 
 '''
 
-class TestNode(SingleNodeConnCB):
+class TestNode(NodeConnCB):
     def __init__(self):
-        SingleNodeConnCB.__init__(self)
+        NodeConnCB.__init__(self)
```
Comment (UdjinM6) [inline test/functional/llmq-is-cl-conflicts.py]:
```suggestion
        super().__init__()
```

File: src/init.cpp (hunk 2 above)
Comment (UdjinM6) [reply src/init.cpp]: Could instead just drop 10086 from the list I guess.

**Comment by PastaPastaPasta** [comment]
dropped 10086, fixed https://github.com/dashpay/dash/pull/2771/commits/6c3d15b459dd95dcd46c01c034c818d6287658eb and force pushed

File: src/init.cpp (hunk 2 above)
Comment (PastaPastaPasta) [reply src/init.cpp]: dropped

**Comment by PastaPastaPasta** [comment]
Travis looks good :)

**Comment by PastaPastaPasta** [comment]
@codablock Please review

**Comment by UdjinM6** [comment]
```
The following scripts are not being run:['txindex.py', 'pruning.py']
Check the test lists in test_runner.py
The command "test "$DEPENDS_TIMEOUT" != "true" && $DOCKER_RUN_IN_BUILDER ./ci/test_integrationtests.sh --jobs=3" exited with 1.
```
🤔

**Comment by PastaPastaPasta** [comment]
:man_shrugging:

**Comment by UdjinM6** [comment]
Pls see https://github.com/UdjinM6/dash/commits/pr2771, tests should finally pass now.

Travis: https://travis-ci.org/UdjinM6/dash/builds/535314818

**Pull request merged**